circuits/types/scalar/src/compare.rs

```diff
@@ -93,37 +83,46 @@ mod tests {
 
     #[test]
     fn test_constant_is_less_than_constant() {
-        check_is_less_than(Mode::Constant, Mode::Constant, 2, 0, 0, 0);
+        check_is_less_than(Mode::Constant, Mode::Constant, 1, 0, 0, 0);
+    }
+
+    #[test]
+    fn test_constant_is_less_than_public() {
+        check_is_less_than(Mode::Constant, Mode::Public, 0, 0, 5, 6);
     }
 
-    // TODO (howardwu): These variate in num_constraints.
-    // #[test]
-    // fn test_constant_is_less_than_public() {
-    //     check_is_less_than(Mode::Constant, Mode::Public, 2, 473, 0, 473);
-    // }
-    //
-    // #[test]
-    // fn test_constant_is_less_than_private() {
-    //     check_is_less_than(Mode::Constant, Mode::Private, 2, 0, 503, 503);
-    // }
+    #[test]
+    fn test_constant_is_less_than_private() {
+        check_is_less_than(Mode::Constant, Mode::Private, 0, 0, 5, 6);
+    }
+
+    #[test]
+    fn test_public_is_less_than_constant() {
+        check_is_less_than(Mode::Public, Mode::Constant, 0, 0, 5, 6);
+    }
+
+    #[test]
+    fn test_private_is_less_than_constant() {
+        check_is_less_than(Mode::Private, Mode::Constant, 0, 0, 5, 6);
+    }
 
     #[test]
     fn test_public_is_less_than_public() {
-        check_is_less_than(Mode::Public, Mode::Public, 2, 0, 1250, 1250);
+        check_is_less_than(Mode::Public, Mode::Public, 0, 0, 5, 6);
     }
 
     #[test]
     fn test_public_is_less_than_private() {
-        check_is_less_than(Mode::Public, Mode::Private, 2, 0, 1250, 1250);
+        check_is_less_than(Mode::Public, Mode::Private, 0, 0, 5, 6);
     }
 
     #[test]
     fn test_private_is_less_than_public() {
-        check_is_less_than(Mode::Private, Mode::Public, 2, 0, 1250, 1250);
+        check_is_less_than(Mode::Private, Mode::Public, 0, 0, 5, 6);
     }
 
     #[test]
     fn test_private_is_less_than_private() {
-        check_is_less_than(Mode::Private, Mode::Private, 2, 0, 1250, 1250);
+        check_is_less_than(Mode::Private, Mode::Private, 0, 0, 5, 6);
     }
```

Comment:
We can write these tests more concisely as `for (mode1, mode2) in [Mode::Private, Mode::Public, Mode::Constant].into_iter().product()`